lib/elix/responders/lists.ex

```diff
@@ -0,0 +1,105 @@
+defmodule Elix.Responders.Lists do
+  @moduledoc """
+  Commands to create, update, and delete one-dimensional lists and their items.
+  """
+
+  use Hedwig.Responder
+  alias Hedwig.Message
+  alias Elix.Lists
+
+  @usage """
+  show lists - Displays all lists
+  """
+  respond ~r/show lists\Z/i, msg do
+    reply(msg, render_items(Lists.all))
+  end
+
+  @usage """
+  create list <name> - Creates a new list with name
+  """
+  respond ~r/create list (.+)/i, %Message{matches: %{1 => list_name}} = msg do
+    Lists.create(list_name)
+    reply(msg, render_items(Lists.all))
+  end
+
+  @usage """
+  show list <list> - Displays the contents of a list by name or number
+  """
+  respond ~r/show list (.+)/i, %Message{matches: %{1 => list_id}} = msg do
+    list_name = parse_list_identifier(list_id)
```

Comment:
[`(?ifthen|else)` groups](http://www.regular-expressions.info/conditional.html) sound like exactly what I need, but unless I'm making a mistake they don't seem to work in Elixir.